R/AbstractGraphReporter.R

```diff
@@ -94,11 +94,21 @@ AbstractGraphReporter <- R6::R6Class(
             )
 
             # Round the double columns to three digits for formatting reasons
-            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+
+            # what are the double columns (but still may contain values with all integers)?
+
+            numDCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+
```

Comment:
I'm a bit confused but this name. What does `numD` mean?

If the goal is "find all columns which are numeric but not integer", I think something like this might be simpler:

```
doubleCols <- Filter(
    x = names(tableObj$x$data)
    , f = function(colName){
         tableObj$x$data[, is.double(get(colName))
    }
)
```

What do you think?

Reply:
Sure, numD refers to the same thing that was previously called numCols which is all Double columns. This is to distinguish between the next variable which are only the Double columns we want to round which actually have true decimal values such as 1.143, 1.743, etc. Sure, the function you have provided their might accomplish the same goal. let me try it out.

Reply:
Regarding this function (I just added comment and closing ] ), I'm getting this error, but as per the comment I think there is a misunderstanding we don't want to simply choose only the columns that are doubles because as I mentioned then the columns which have all integer-like values will be rounded with 3 zeros as per my earlier comment.

```
    # Round only the (non-integer coercible) double columns to three digits
        doubleCols <- Filter(
                x = names(tableObj$x$data)
               ,  f = function(colName){
                           tableObj$x$data[, is.double(get(colName))]
             }
            )

            tableObj <- DT::formatRound(
                columns = doubleCols
                , table = tableObj
                , digits = 3
            )
            return(tableObj)
```
Error message:
```
## Error in get(colName): object 'node' not found
```
That's understandable as I realize you just gave an idea for me to try out.

Reply:
I'm open to renaming the variables if you have a more intuitive name (assuming you are following my logic of why I think I have to check if each value in a double column is coercible to integer, because if not all of them are coercible that is the only situation where we need to round).

Reply:
Yes, you have stated it correctly as the goal is: "find all columns which are numeric but not integer". I'm open to simpler approaches but couldn't find one. However, I believe it is generalizable as I'm first filtering only double columns and then applying this function to only those double columns.

Reply:
Ah ok I see. Yes apologies, that was hastily-typed code that I didn't actually test.

I'm not sure I agree that "could be coerced to an integer" is desirable here. I think it is ok to have `1.00, 2.00, 3.00` etc. and to say "if you want these to be formatted as integers, cast them as integers when creating the data".

Otherwise, we will get situations such as this:

1. User uses `FunctionReporter` and adds test coverage to the table. User has only 1 test that covers 10 of the 40 lines in the codebase, so `coverage` column is formatted as `25`
2. User adds a unit test, getting coverage up to 15 of the 40 lines. `coverage` column is formatted as `37.500`.

In my opinion this behavior is not desirable and the decision of whether the data is truly double or integer should be pushed up to the code that generates the data frame, not this visualization code

Reply:
The variable names are ok! I just wanted to be sure I understood.

Thank you for being patient with me 😄

Reply:
No problem. I agree its a better approach to not have that type of data come in as a Double and be integer when read in. The only challenge is that since the data is currently all Double I don't know how to accomplish this without manually changing certain columns to integers which is a worse approach.

Reply:
Actually let me have another look to try to find the part of the code that generates the data frame and see if there is a generalizable approach to specify certain columns to integers.

Reply:
Two integers I can think of off the top of my head (that we might have failed to cast as integers): `inDegree` and `outDegree`. Those would be produced by the individual reporters, I think.

Reply:
I'm having trouble locating the code that produces the reporters could you point me in the right direction? All but 2 of the numeric/double columns were identified as integer as per my earlier approach. They had num something in the column name.

Reply:
There are some [here](https://github.com/uptake/pkgnet/blob/master/R/GraphClasses.R#L357,L366) for example that might be cast to integer.

I am actually stepping away from my computer shortly to begin traveling, but @jayqi or @bburns632 will be able to pick this up at some point and help you out. Thanks for sticking with it!